### PR TITLE
Refactor cai2hcl/ structure

### PR DESCRIPTION
**Pull request by amirkaromashkin**
Goal: Refactor file structure to make it compatible with MMv1 generation.
This PR is a prerequisite for [PR which moves TGC/cai2hcl ownership to MM](https://github.com/GoogleCloudPlatform/magic-modules/pull/8824)

Changes:

1. Move each converter to a decided service package.
Move existing converters to "compute" package.
Extract "common" package to avoid circular dependency: cai2hcl(convert.go) -> compute(compute_instance) -> cai2hcl(utils.go). Move resources to a folder "conveters/google/resources" using same folder pattern as tfplan2cai has.

4. Move testdata from the root folder inside cai2hcl/.
tfplan2cai/ has its own test data - the only test which used common test data was redirected to internal copy of the test file empty-0.13.7.tfplan.json.
tfplan2cai and cai2hcl codegen is implemented as 2 different providers in MMv1, so it makes sense to split there tests.
This will make codegen easier as each one will update only its folder. If in future tfplan2cai and cai2hcl converters are merged together, testdata can be moved to a shared folder.
At the current early stage of cai2hcl converters, merging 2 providers may cause more issues than benefits (i.e. 2-way converter tests)/

### CODE REVIEW

File: cai2hcl/converters/google/resources/project.go [hunk 1]
```diff
@@ -1,10 +1,11 @@
-package cai2hcl
+package resources
```
Comment (iyabchen) [inline cai2hcl/converters/google/resources/project.go:1]: why project is not in a separate folder like compute?

Comment (amirkaromashkin) [reply cai2hcl/converters/google/resources/project.go:1]: This file combines 2 APIs: cloudbilling and cloudresourcemanager. I am not sure where to put it

Comment (iyabchen) [reply cai2hcl/converters/google/resources/project.go:1]: I feel it's safe to put cloudbilling into the compute package, in my memory, cloud billing can only live with the project object.

Comment (amirkaromashkin) [reply cai2hcl/converters/google/resources/project.go:1]: You mean into the cloudresourcemanager package?

Comment (melinath) [reply cai2hcl/converters/google/resources/project.go:1]: these should go in a folder called `resourcemanager`, at least for now - this billing info is part of the google_project resource, not a separate resource.

**Review by melinath — CHANGES_REQUESTED** [review]
The structure of cai2hcl should be more similar to the [provider](https://github.com/hashicorp/terraform-provider-google-beta/tree/main/google-beta) than to tfplan2cai - the `converters/google/resources` folder structure is a historical artifact that we are currently working to move away from.

We should aim for something like:

```
- cai2hcl
   |- convert
       |- convert.go
       |- convert_test.go
       |- converter_map.go
   |- common
       |- converter.go
       |- utils.go
   |- services
      |- compute
          |- compute_forwarding_rule.go
          |- compute_forwarding_rule_test.go
          |- compute_instance.go
          |- compute_instance_test.go
      |- resourcemanager
          |- project.go
          |- project_test.go
```

File: cai2hcl/converters/google/resources/project.go (hunk 1 above)
Comment (amirkaromashkin) [reply cai2hcl/converters/google/resources/project.go]: Done

**Comment by amirkaromashkin** [comment]
> The structure of cai2hcl should be more similar to the [provider](https://github.com/hashicorp/terraform-provider-google-beta/tree/main/google-beta) than to tfplan2cai - the `converters/google/resources` folder structure is a historical artifact that we are currently working to move away from.
> 
> We should aim for something like:
> 
> ```
> - cai2hcl
>    |- convert
>        |- convert.go
>        |- convert_test.go
>        |- converter_map.go
>    |- common
>        |- converter.go
>        |- utils.go
>    |- services
>       |- compute
>           |- compute_forwarding_rule.go
>           |- compute_forwarding_rule_test.go
>           |- compute_instance.go
>           |- compute_instance_test.go
>       |- resourcemanager
>           |- project.go
>           |- project_test.go
> ```

Done all except the "convert" package. I believe it should be left it as is due to 2 reasons:
1. cai2hcl API uses path to cai2hcl/ package as a main entry point importing both Convert function and Options structure. Such change will break the API (or will need some kind of delegation).
2. Root of the folder is a good place for the library entry point, these 3 files are good start for initial discovery of the library.

**Review by melinath — CHANGES_REQUESTED** [review]
> Done all except the "convert" package. I believe it should be left it as is due to 2 reasons:
> 
> 1. cai2hcl API uses path to cai2hcl/ package as a main entry point importing both Convert function and Options structure. Such change will break the API (or will need some kind of delegation).
> 2. Root of the folder is a good place for the library entry point, these 3 files are good start for initial discovery of the library.

SGTM. Could you resolve the merge conflicts?

**Review by melinath — CHANGES_REQUESTED** [review]
One more change but otherwise LGTM

File: cai2hcl/convert_test.go [hunk 2]
```diff
@@ -1,18 +1,57 @@
...
 
-func assertTestData(fileName string) (err error) {
-	assetFilePath := fmt.Sprintf("../testdata/%s.json", fileName)
-	expectedTfFilePath := fmt.Sprintf("../testdata/%s.tf", fileName)
+type TestCase struct {
+	name         string
+	sourceFolder string
+}
+
+var testDataFileNames = []string{
```
[the remark this answers is not in the document]
Comment (amirkaromashkin) [reply cai2hcl/convert_test.go]: I like this, yeah, done.
Still this required some refactoring and which needs a review. PTAL

**Review by melinath — APPROVED** [review]

**Pull request merged**